usecase/interactor/pokeinteractor.go

```diff
@@ -0,0 +1,149 @@
+package interactor
+
+import (
+	"errors"
+	"fmt"
+	"github.com/ToteEmmanuel/academy-go-q12021/domain/model"
+	"github.com/ToteEmmanuel/academy-go-q12021/usecase/presenter"
+	"github.com/ToteEmmanuel/academy-go-q12021/usecase/repository"
+	"github.com/go-resty/resty/v2"
+	"sync"
+)
+
+type pokeInteractor struct {
+	PokeRepo      repository.PokeRepository
+	PokePresenter presenter.PokePresenter
+	RestClient    *resty.Client
+}
+
+type PokeInteractor interface {
+	Get(id int32) (*model.Pokemon, error)
+	GetAll() ([]*model.Pokemon, error)
+	CatchOne(int32) (*model.Pokemon, error)
+}
+
+func NewPokeInteractor(r repository.PokeRepository, p presenter.PokePresenter, client *resty.Client) PokeInteractor {
+	return &pokeInteractor{
+		PokeRepo:      r,
+		PokePresenter: p,
+		RestClient:    client,
+	}
+}
+
+func (pI *pokeInteractor) Get(id int32) (*model.Pokemon, error) {
+	p, err := pI.PokeRepo.FindById(id)
+	if err != nil {
+		return nil, err
+	}
+	return pI.PokePresenter.ResponsePokemon(p), nil
+}
+
+func (pI *pokeInteractor) GetAll() ([]*model.Pokemon, error) {
+	p, err := pI.PokeRepo.FindAll()
+	if err != nil {
+		return nil, err
+	}
+	return pI.PokePresenter.ResponsePokemons(p), nil
+}
+
+func (pI *pokeInteractor) CatchOne(id int32) (*model.Pokemon, error) {
+	p, err := pI.lookForPokemon(id)
+	if err != nil {
+		return nil, err
+	}
+	return pI.PokePresenter.ResponsePokemon(p), nil
+}
+
+func (pI *pokeInteractor) lookForPokemon(id int32) (*model.Pokemon, error) {
+	var waitgroup sync.WaitGroup
+	waitgroup.Add(2)
+	catchedPokemon := &model.Pokemon{
+		Id:         id,
+		Species:    "",
+		Sprite:     "",
+		FlavorText: "",
+		Types:      nil,
+	}
+	var err error
+	err = fetchBase(catchedPokemon, pI.RestClient)
+	if err != nil {
+		fmt.Println(err)
```

Comment:
Is it ok if you just log the error and not return here?

Reply:
Changed the way the error is handled with a channel and an error response.